packages/svelte/src/internal/client/runtime.js

```diff
@@ -409,7 +414,17 @@ export function update_reaction(reaction) {
 	skipped_deps = 0;
 	untracked_writes = null;
 	active_reaction = (flags & (BRANCH_EFFECT | ROOT_EFFECT)) === 0 ? reaction : null;
-	skip_reaction = !is_flushing_effect && (flags & UNOWNED) !== 0;
+	// prettier-ignore
+	skip_reaction =
+		(flags & UNOWNED) !== 0 &&
+		(!is_flushing_effect ||
+			// If we were previously not in a reactive context and we're reading an unowned derived
+			// that was created inside another derived, then we don't fully know the real owner and thus
+			// we need to skip adding any reactions for this unowned
+				((previous_reaction === null || previous_untracking) &&
+				(/** @type {Derived} */ (reaction).parent !== null &&
+				(/** @type {Derived} */ (reaction).parent.f & DERIVED) !== 0)));
```

Comment:
Pretty sure you can reduce this to `/** @type {Derived} */ (reaction).parent?.f & DERIVED) !== 0`

Reply:
You can't use optional chaining with a bitwise flag?

Reply:
Of course you can?
```js
let foo = { x: { i: 2 } };
console.log((foo.x.i & 2) !== 0); // true
console.log((foo.y?.i & 2) !== 0); // false
```
it doesn't work in the case of `=== 0` because then regardless of undefined or a number it's always 0, but it workds in the `!== 0` case